test/integration/oauth_token_endpoint_test.go

```diff
@@ -0,0 +1,158 @@
+package integration
+
+import (
+	"crypto/tls"
+	"net/http"
+	"net/url"
+	"testing"
+
+	"golang.org/x/net/html"
+
+	knet "k8s.io/apimachinery/pkg/util/net"
+
+	configapi "github.com/openshift/origin/pkg/cmd/server/api"
+	testutil "github.com/openshift/origin/test/util"
+	testserver "github.com/openshift/origin/test/util/server"
+)
+
+// TestOAuthRequestTokenEndpoint tests obtaining and using a bearer token from the request and display token endpoints.
+func TestOAuthRequestTokenEndpoint(t *testing.T) {
+	// Set up server
+	masterOptions, err := testserver.DefaultMasterOptions()
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	defer testserver.CleanupMasterEtcd(t, masterOptions)
+
+	masterOptions.OAuthConfig.IdentityProviders[0] = configapi.IdentityProvider{
+		Name:            "test",
+		UseAsChallenger: true,
+		UseAsLogin:      true,
+		MappingMethod:   "claim",
+		Provider:        &configapi.AllowAllPasswordIdentityProvider{},
+	}
+	clusterAdminKubeConfig, err := testserver.StartConfiguredMaster(masterOptions)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	clientConfig, err := testutil.GetClusterAdminClientConfig(clusterAdminKubeConfig)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	transport := knet.SetTransportDefaults(&http.Transport{
+		TLSClientConfig: &tls.Config{
+			InsecureSkipVerify: true,
+		},
+	})
+
+	// Hit the token request endpoint
+	masterURL, err := url.Parse(clientConfig.Host)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	masterURL.Path = "/oauth/token/request"
+	tokenReq, err := http.NewRequest("GET", masterURL.String(), nil)
+	if err != nil {
+		t.Fatalf("unexpected error %v", err)
+	}
+	tokenResp, err := transport.RoundTrip(tokenReq)
+	if err != nil {
+		t.Fatalf("unexpected error %v", err)
+	}
+	defer tokenResp.Body.Close()
+
+	if tokenResp.StatusCode != http.StatusFound {
+		t.Fatalf("unexpected response code %v", tokenResp.StatusCode)
+	}
+
+	// Follow location header to auth
+	authRedirect, err := url.Parse(tokenResp.Header.Get("Location"))
```

Comment:
I think you want to check to make sure you have something in the location header.